bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Observation.scala

```diff
@@ -1,36 +1,43 @@
 package edu.gemini.model.p1.immutable
 
+import edu.gemini.model.p1.overheads.Overheads
 import edu.gemini.model.p1.{mutable => M}
+
 import scalaz._
 import Scalaz._
 
 object Observation {
 
-  // Some lenses
+  // Some lenses.
   val blueprint:Lens[Observation, Option[BlueprintBase]] = Lens.lensu((a, b) => a.copy(blueprint = b), _.blueprint)
   val condition:Lens[Observation, Option[Condition]] = Lens.lensu((a, b) => a.copy(condition = b), _.condition)
   val target:Lens[Observation, Option[Target]] = Lens.lensu((a, b) => a.copy(target = b), _.target)
   val meta:Lens[Observation, Option[ObservationMeta]] = Lens.lensu((a, b) => a.copy(meta = b), _.meta)
-  val time:Lens[Observation, Option[TimeAmount]] = Lens.lensu((a, b) => a.copy(time = b), _.time)
+  val intTime:Lens[Observation, Option[TimeAmount]] = Lens.lensu((a, b) => a.copy(intTime = b), _.intTime)
 
   def apply(blueprint:Option[BlueprintBase],
             condition:Option[Condition],
             target:Option[Target],
             band:Band,
-            time:Option[TimeAmount]) = new Observation(blueprint, condition, target, time, band)
+            intTime:Option[TimeAmount],
+            progTime:Option[TimeAmount] = None) = new Observation(blueprint, condition, target, intTime, progTime, band)
```

Comment:
The reason is primarily in the migration.

Old proposals will have a `progTime` (and not a fully calculated `ObservationTimes`) and need to calculate the `intTime` from this, and then use this `intTime` to calculate the full `ObservationTimes`.

New 2017B and onward proposals will have an `intTime` and will use this to calculate the full `ObservationTimes`.

My initial thought was actually to pass in an `Option[ObservationTimes]` as you say, but as the only field with a valid value would ever be `progTime`, I thought this might be the wrong approach.

I'm open to alternative suggestions. 😸